src/Rendering/prePassRenderer.ts

```diff
@@ -178,47 +216,32 @@ export class PrePassRenderer {
         this._engine = scene.getEngine();
 
         PrePassRenderer._SceneComponentInitialization(this._scene);
-        this._resetLayout();
+        this.defaultRT = this._createRenderTarget("sceneprePassRT", null);
+        this._setRenderTarget(null);
     }
 
-    private _initializeAttachments() {
-        const multiRenderLayout = [];
-        const clearLayout = [false];
-        const defaultLayout = [true];
-
-        for (let i = 0; i < this.mrtCount; i++) {
-            multiRenderLayout.push(true);
-
-            if (i > 0) {
-                clearLayout.push(true);
-                defaultLayout.push(false);
-            }
-        }
-
-        this._multiRenderAttachments = this._engine.buildTextureLayout(multiRenderLayout);
-        this._defaultAttachments = this._engine.buildTextureLayout(defaultLayout);
-    }
+    /**
+     * Creates a new PrePassRenderTarget
+     * This should be the only way to instanciate a `PrePassRenderTarget`
+     * @param name Name of the `PrePassRenderTarget`
+     * @param renderTargetTexture RenderTarget the `PrePassRenderTarget` will be attached to.
+     * Can be `null` if the created `PrePassRenderTarget` is attached to the scene (default framebuffer).
+     * @hidden
+     */
+    public _createRenderTarget(name: string, renderTargetTexture: Nullable<RenderTargetTexture>) : PrePassRenderTarget {
+        const rt = new PrePassRenderTarget(name, renderTargetTexture, { width: this._engine.getRenderWidth(), height: this._engine.getRenderHeight() }, 0, this._scene,
+            { generateMipMaps: false, generateStencilBuffer: true, defaultType: Constants.TEXTURETYPE_UNSIGNED_INT, types: [], drawOnlyOnFirstAttachmentByDefault: true });
 
-    private _createCompositionEffect() {
-        this.prePassRT = new MultiRenderTarget("sceneprePassRT", { width: this._engine.getRenderWidth(), height: this._engine.getRenderHeight() }, this.mrtCount, this._scene,
-            { generateMipMaps: false, generateDepthTexture: true, defaultType: Constants.TEXTURETYPE_UNSIGNED_INT, types: this._mrtFormats });
-        this.prePassRT.samples = 1;
+        this.renderTargets.push(rt);
 
-        this._initializeAttachments();
-        if (this._useGeometryBufferFallback && !this._geometryBuffer) {
-            // Initializes the link with geometry buffer
-            this.useGeometryBufferFallback = true;
-        }
-
-        this.imageProcessingPostProcess = new ImageProcessingPostProcess("sceneCompositionPass", 1, null, undefined, this._engine);
-        this.imageProcessingPostProcess.autoClear = false;
+        return rt;
     }
 
     /**
      * Indicates if rendering a prepass is supported
      */
     public get isSupported() {
-        return this._scene.getEngine().getCaps().drawBuffersExtension;
+        return this._engine.webGLVersion > 1 || this._scene.getEngine().getCaps().drawBuffersExtension;

```

Comment:
Please remove `this._engine.webGLVersion > 1`, it won't work in WebGPU.

`this._scene.getEngine().getCaps().drawBuffersExtension` should be enough for the pre pass renderer to work? Is there another feature that would not make it work in webgl 1?

Reply:
subsurfacescattering needs webgl2, but now since prepass is uncorrelated from it, we can remove the webGL2 check